mitodl/mit-open
# 130: Add custom logout and redirect to keycloak

# What are the relevant tickets?
https://github.com/mitodl/mit-open/issues/130

# Description (What does it do?):
When a user visits http://open.ol.local:8063/logout:

1. They are logged out of MIT Open.
2. They are redirected to the Keycloak logout endpoint and automatically logged out of Keycloak.

This PR makes sure that a user can logout of MIT Open and Keycloak at the same time.  In the future this functionality will be used by Keycloak to log the user out of any other Open Learning applications the user is currently logged in to.

https://docs.google.com/document/d/17tJ-C2EwWoSpJWZKjuhMVgsqGtyPH0IN9KakXvSKU0M/edit#heading=h.9y6s0jvyqjsw

2 environment variables have been added and are required:
`KEYCLOAK_BASE_URL`
`KEYCLOAK_REALM_NAME`

Instructions for setting the Keycloak related environment variables that are required have been added to the README for this repository.

# How can this be tested?
These instructions assume that you have access to an existing Keycloak account or can create a new Keycloak account at https://sso-qa.odl.mit.edu/.

1. Configure the required Keycloak environment variables described in the README based on the `ol-open-discussions-local` OpenID Connect client configured in Keycloak in the QA environment.
2. Start the MIT Open application.
4. Visit http://open.ol.local:8063/login/ol-oidc and login via your Keycloak account.
5. Verify that you are logged in to MIT Open (can be verified by visiting http://open.ol.local:8063/admin).
6. Verify that you are logged in to Keycloak in the QA environment (can be verified by visiting https://sso-qa.odl.mit.edu/realms/olapps/account/ and not being asked to login)
7. Visit http://open.ol.local:8063/logout.
8. Verify that you are logged out of MIT Open (can be verified by visiting http://open.ol.local:8063/admin).
9. Verify that you are logged out of Keycloak in the QA environment (can be verified by visiting https://sso-qa.odl.mit.edu/realms/olapps/account/ and seeing the login screen).
